backend/services/job.js

```diff
@@ -376,11 +383,6 @@ export async function completeJob(jobId, userId) {
   const job = await JobModel.findById(jobId);
   if (!job) throw ServiceError.JOB_NOT_FOUND;
 
-  // Validate client job ownership
-  if (job.client !== userId) {
-    throw ServiceError.JOB_EDIT_PERMISSION_DENIED;
```

Comment:
Is there a reason we're deleting this ownership validation check?

Reply:
So because we need to mark the job as completed from the driver side, we have to delete the check. I assume this code was here before because the person writing this thought that the farmer would complete the job. I'll replace this check with `job.assignedDriver` instead of `job.client` though